### PR TITLE
Add 6.11.0 release notes

### PR DESCRIPTION
**Pull request by rbuch**
Please review and add things I may have missed or undersold.

### CODE REVIEW

File: CHANGES [hunk 1]
```diff
@@ -1,6 +1,111 @@
...
+- Fixed tlsglobals migration on macOS.
+
+- Fixed bugs in activation of migration callbacks.
+
+- Fixed a bug in MPI_Waitsome.
+
+- Accept arguments from AMPI_BUILD_FLAGS environment variable for
+  ampiCC.
+
+- Added PUP methods for derived datatypes.
```
Comment (stwhite91) [inline CHANGES:105]: This doesn't need to be in the release notes, it's a minor implementation detail

File: CHANGES [hunk 2]
```diff
@@ -1,6 +1,111 @@
...
+- Bounded number of hops messages can take when chares frequently
+  migrate.
+
+Adaptive MPI:
+
+- Isomalloc sync is now enabled by default and the implementation no
+  longer uses the filesystem to pass data. Isomalloc sync can be
+  disabled with +no_isomalloc_sync.
+
+- Added AMPI-only build target, AMPI-only.
```
Comment (stwhite91) [inline CHANGES:94]: I'd add that this build optimizes AMPI by disabling features of Charm++ that AMPI doesn't use.

File: CHANGES [hunk 3]
```diff
@@ -1,6 +1,111 @@
...
+
+- Support for variable sized messages in TRAM.
+
+- Added a pup_buffer API with zero copy functionality.
+
+- Array broadcasts are now node aware, avoiding unnecessary
+  duplication of messages. Expedited nokeep array broadcasts are also
+  allowed now.
+
+- Added CmiNodeReduce API for node level reductions.
```
Comment (stwhite91) [inline CHANGES:49]: Maybe "Converse node level reductions"

File: CHANGES [hunk 4]
```diff
@@ -1,6 +1,111 @@
...
+================================================================================
+What's new in Charm++ 6.11.0
+================================================================================
+
+This is a feature release, with the following major changes:
+
+Misc:
+
+- Charm++ now builds with CMake by default when using the `./build`
+  script (requires version 3.4 or higher). The old build system is
```
Comment (matthiasdiener) [inline CHANGES:13]:
```suggestion
  script (requires CMake version 3.4 or higher). The old build system is
```

File: CHANGES [hunk 5]
```diff
@@ -1,6 +1,111 @@
...
+  https://charm.readthedocs.io/en/latest/charm++/manual.html#installation-with-cmake
+  for more information.
+
+- We have adopted GitHub discussions
+  (https://github.com/UIUC-PPL/charm/discussions) as the preferred
+  venue for Charm++ questions and discussions instead of the
+  charm@cs.illinois.edu mailing list.
+
+- The BigSim emulation facility has been removed from the runtime
+  system.
```
Comment (matthiasdiener) [inline CHANGES:24]:
```suggestion
- The BigSim emulator has been removed from Charm++.
```

Comment (matthiasdiener) [reply CHANGES:24]: Is there any other simulator that we can recommend as a substitute?

File: CHANGES [hunk 6]
```diff
@@ -1,6 +1,111 @@
...
+
+- A significant overhaul of the load balancing infrastructure and the
+  addition of TreeLB, a new, more flexible and extensible base LB,
+  intended to replace the previous CentralLB and HybridBaseLB designs
+  (load balancers using these previous designs are still
+  supported). Some included strategies have been rewritten to use
+  TreeLB, unused and extraneous strategies have been removed. Please
+  see
+  https://charm.readthedocs.io/en/latest/charm++/manual.html#load-balancing
+  for more information.
```
Comment (matthiasdiener) [inline CHANGES:36]: I think this should be highlighted earlier (it's not only relevant for Charm++).

File: CHANGES (hunk 5 above)
Comment (rbuch) [reply CHANGES]: Potentially https://computing.llnl.gov/projects/interconnection-networks/network-modeling-and-simulation, but I'm not sure if that's really a complete replacement.

File: CHANGES [hunk 7]
```diff
@@ -1,6 +1,111 @@
...
+
+- Support for intra-node GPU messages, allowing data to be sent
+  to/from GPU memory without touching host memory.
```
Comment (minitu) [inline CHANGES]:
```suggestion
- Documentation for GPU support in Charm++ is now located at 
  https://charm.readthedocs.io/en/latest/charm++/manual.html#gpu-support. 

- Support for persistent intra-node GPU messages, and experimental support for
  regular (nonpersistent) intra-node GPU messages. This feature allows data to be
  sent to/from GPU memory without going through host memory.
```
Persistent intra-node GPU messaging is currently being implemented with high priority here: https://github.com/UIUC-PPL/charm/pull/3101

File: CHANGES [hunk 8]
```diff
@@ -1,6 +1,115 @@
...
+  (load balancers using these previous designs are still
+  supported). Some included strategies have been rewritten to use
+  TreeLB, unused and extraneous strategies have been removed. Please
+  see
+  https://charm.readthedocs.io/en/latest/charm++/manual.html#load-balancing
+  for more information.
+
+- Support for persistent intra-node GPU messages, and experimental support for
+  regular (nonpersistent) intra-node GPU messages. This feature allows data to be
+  sent to/from GPU memory without going through host memory. Please
```
Comment (minitu) [inline CHANGES:24]:
```suggestion
- Experimental support for intra-node GPU messages, allowing data to be
  sent to/from GPU memory without going through host memory. Please
```

**Review by matthiasdiener — APPROVED** [review]

**Review by minitu — APPROVED** [review]

**Pull request merged**